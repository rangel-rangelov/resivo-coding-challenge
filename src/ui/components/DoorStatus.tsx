import Typography from "@mui/material/Typography";
import { ConnectionStatus } from "@/models/ConnectionStatus";

interface DoorStatusProps {
  status: ConnectionStatus;
}

export function DoorStatus({ status }: DoorStatusProps) {
  return (
    <Typography
      color={`${status === ConnectionStatus.Online ? 'success' : 'error'}.main`}
    >
      {status}
    </Typography>
  )
};
