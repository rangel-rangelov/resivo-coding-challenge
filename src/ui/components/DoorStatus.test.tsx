import { render } from '@testing-library/react';
import { ConnectionStatus } from '@/models/ConnectionStatus';
import { DoorStatus } from './DoorStatus';

jest.mock('next/router', () => require('next-router-mock'));

describe('DoorStatus', () => {
  it('should render correctly', () => {
    const { container } = render(<DoorStatus status={ConnectionStatus.Online} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
