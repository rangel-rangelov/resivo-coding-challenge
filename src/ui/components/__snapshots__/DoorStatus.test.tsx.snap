// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DoorStatus should render correctly 1`] = `
<p
  class="MuiTypography-root MuiTypography-body1 css-18todbp-MuiTypography-root"
>
  online
</p>
`;
